db/migrate/20220921095507_add_coupon_type_and_percentage_rate_to_coupons.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class AddCouponTypeAndPercentageRateToCoupons < ActiveRecord::Migration[7.0]
+  def change
+    add_column :coupons, :coupon_type, :integer, null: false, default: 0
+    add_column :coupons, :percentage_rate, :decimal, precision: 10, scale: 5
```

Comment:
I was thinking to have more precision and handle it the same way as wallet's rate, since we are dealing with money. We can also later adjust precision and scale if needed... Do you think that we should switch to float?

Reply:
Okay, make sense :)

It's good. Maybe we will have to migrate existing float fields to this datatype